fabric-crash-report-info-v1/src/main/java/net/fabricmc/fabric/mixin/crash/report/info/MixinCrashReport.java

```diff
@@ -39,22 +39,35 @@ public abstract class MixinCrashReport {
 	@Inject(at = @At("RETURN"), method = "<init>")
 	private void fillSystemDetails(CallbackInfo info) {
 		addSection("Fabric Mods", () -> {
-			Map<String, String> mods = new TreeMap<>();
+			TreeSet<ModContainer> topLevelMods = new TreeSet<>(Comparator.comparing(mod -> mod.getMetadata().getId()));
```

Comment:
I recommend adding to an array list and sorting them later, as a tree structure is better if you query while the structure can update. Tree sets elements are linked, meaning they are less optimizable by jvm compared to arrays allocated in consecutive memory blocks.